### DISCUSSION TITLE
[Idea] Add two hooks to extends access control for product and category in FO

### DISCUSSION THREAD

**Discussion by jf-viguier** [PrestaShop/PrestaShop#32985] [Ideas]
### The problem

Product::checkAccessStatic and Caterogy::checkAccess don't have any hook to extend the access control behaviour.

### The solution

Add two hooks to avoid overriding this two methods.

### Do you plan to work on this subject?

- [ ] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [ ] I'm willing to submit a Pull Request that implements this feature.
- [ ] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by kpodemski** [comment]
Sounds good. Go for it 🎉 😄
